### ISSUE TITLE
apply_nbeats_patch removed without deprecation; LayerNorm now applied unconditionally — please confirm intent

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-r2darts2#36]
Two questions about the same edit — both about behaviour rather than packaging.

## 1. `apply_nbeats_patch` was removed with no deprecation path

In 0.2.1, `views_r2darts2/__init__.py` uses a module-level `__getattr__` with an explicit allowlist.
`apply_nbeats_patch` is in neither `__all__` nor the dispatch chain, so

```python
from views_r2darts2 import DartsForecastingModelManager, apply_nbeats_patch
```

raises `AttributeError` **at import time**, before anything else runs. Four models in `views-models`
(`emerging_principles`, `novel_heuristics`, `preliminary_directives`, `shining_codex`) imported and
called it in `main.py`.

The reason it was load-bearing rather than incidental: at 0.1.1, `apply_all_patches()` had
`# apply_nbeats_patch()` **commented out**, so those four explicit calls were the only route by which
that patch was ever applied anywhere.

## 2. `apply_nbeats_layernorm_patch` is not a drop-in replacement

The name similarity is misleading, and I want to flag that before anyone (including us) swaps one for
the other on a name match:

| | 0.1.1 `apply_nbeats_patch` | 0.2.1 `apply_nbeats_layernorm_patch` |
|---|---|---|
| patches | `_Block.__init__` **and** `_Block.forward` | `_Block.forward` and `_Stack.forward` |
| what it does | swaps `nn.Dropout` → `MonteCarloDropout` in `dropout_stack`, keeping dropout live at inference | inserts functional LayerNorm after each linear, and after each residual subtraction |
| concern | **uncertainty** | **normalisation** |
| in `apply_all_patches()` | no — commented out | **yes, unconditionally** |

These solve different problems. Substituting one for the other silently changes what a model computes.

**Ask (a):** was dropping the MonteCarloDropout behaviour deliberate? Specifically — did darts 0.46.1's
own `_Block` absorb it, making the patch redundant? If so that's worth a line in the release notes,
because from the outside it reads as an accidental removal.

**Ask (b):** `apply_all_patches()` is called unconditionally from `DartsForecastingModelManager.__init__`,
and now includes the LayerNorm patch. That applies LayerNorm to **every** NBEATS/NHiTS model, including
those that never opted into any patch. Is that intended? It's an un-opted-in numerical change for
models whose configs say nothing about it, and it won't show up as an error — only as different
numbers.

## A note in your favour

A straight port of the old patch would not have worked anyway. It iterates `self.fc_stack` /
`self.bn_stack` / `self.dropout_stack`; the 0.2.1 patch iterates `self.linear_layer_stack_list`. darts
restructured `_Block`'s internals between 0.40.0 and 0.46.1, so keeping the symbol alive would have
required rewriting it, not just retaining it. That makes the removal understandable — it's the silence
around it that costs consumers.

## Impact on us — small, and already absorbed

For completeness, so this isn't read as more urgent than it is. We've removed the four calls on our
side. Of the four models, all declare `num_samples: 1`, and two of them (`novel_heuristics`,
`shining_codex`) had `mc_dropout: False` — they were calling a patch they had switched off. So the
forecast consequence is close to nil.

Ironically the one model where MC dropout would actually have mattered — `cold_heart`, NBEATS with
`num_samples: 512` and `mc_dropout: True` — never called the patch at all. That's ours to sort out, not
yours; mentioning it only so the priority here reads correctly. This is a "please confirm the intent"
issue, not a "this broke production" one.

Related: #34 (the resolution conflicts), #35 (predstore).

**Comment by smellycloud** [comment]
Patch breaks updated nbeats code from darts iirc. Dropout still seems to work regardless. LayerNorm was more pressing because training was being destabilized when distributions shift over time periods and entities. Same patch was applied to NHiTS.

**Comment by Polichinel** [comment]
Thanks, that's the context I was missing — the distribution-shift instability makes the LayerNorm patch a clear win, and it's good to know NHiTS got the same treatment. One small ask so it stays safe: could the patch module state which Darts version it was written against, and maybe assert it at import? The docstring still says 0.45 while we pin 0.46.1, and a monkey-patch on a private class is exactly the thing a Darts bump would break without a message. No rush; happy to leave this open as a reminder or close it if you'd rather track it elsewhere.

**Comment by smellycloud** [comment]
for now i am pinning darts as my safeguard. i suspect that i will find more issues over time so i'd rather not keep chasing darts updates for now and instead maintain a stable base that i periodically bump up after testing the whole thing end-to-end. i will update the docs eventually.
